src/test/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSenderTest.java

```diff
@@ -0,0 +1,119 @@
+package com.ericsson.gerrit.plugins.eiffel.messaging;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
+import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.EiffelEventSenderException;
+
+public class EiffelEventSenderTest {
+
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
+    private ResponseEntity response;
+
+    private static final String EIFFEL_TYPE = "EiffelSourceChangeCreatedEvent";
+    private static final int STATUS_OK = HttpStatus.SC_OK;
+    private static final int STATUS_NOT_FOUND = HttpStatus.SC_NOT_FOUND;
+
+    @Before
+    public void beforeTest() throws IOException, URISyntaxException {
+        setUpMockObjects();
+    }
+
+    @Test
+    public void testEventSender() throws IOException, URISyntaxException {
+        setUpMockActions();
+
+        EiffelEventSender sender = new EiffelEventSender(pluginConfig, httpRequest);
+        sender.setMessage(new EiffelSourceChangeCreatedEvent());
+        sender.setType(EIFFEL_TYPE);
+
+        try {
+            Whitebox.invokeMethod(sender, "generateAndPublish");
+        } catch (Exception e) {
+            String exceptionType = e.getClass().getSimpleName();
+            String assertError = String.format(
+                    "No exception should have been thrown but was %s",
+                    exceptionType);
+            fail(assertError);
+        }
+    }
+
+    @Test
```

Comment:
Is there a reason not to use the expected exception or rule tag (https://stackoverflow.com/questions/15216438/junit-testing-exceptions?noredirect=1&lq=1 - 1st or 2nd answer)

Reply:
Yes I know about expected in the Test annotation and I've used at least on one occasion before. I can verify that the Exception happened for the right reason if I catch the exception myself.

Reply:
What I mean with right reason is that the exception one that can occur in two different scenarios. If I had a single type of exception for each scenario than we'd be sure without having to check further. Perhaps that is more desirable?

Reply:
This one I'm not sure about, perhaps I'll make two separate exceptions so there doesn't need to be more than an expected exception.

Reply:
I added separate exceptions and removed all asserts and let the exception throws decide if a test has succeeded or not.